### PR TITLE
EXOGTN-2224 New portlet added in a page via UI fail to render

### PR DESCRIPTION
**Pull request by boubaker**
Make the dot in application id considered inside application name.

### CODE REVIEW

File: api/src/main/java/org/gatein/pc/api/PortletContext.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ private Components interpretIntoComponents(String id)
          if (trimmedId.startsWith(PREFIX))
          {
             // check the case: /application.portlet
-            int separator = trimmedId.indexOf(SEPARATOR); // find first separator, other separator are considered part of the portlet name
+            int separator = trimmedId.lastIndexOf(SEPARATOR); // find first separator, other separator are considered part of the portlet name
```
Comment (thomasdelhomenie) [inline api/src/main/java/org/gatein/pc/api/PortletContext.java]: I think it will be a problem if the portlet name contains a dot.

Comment (boubaker) [inline api/src/main/java/org/gatein/pc/api/PortletContext.java]: Yes, either the portlet name that contains dot or the application name that contains dot. The current implementation doesn't support both.

**Comment by thomasdelhomenie** [comment]
Won't this change break backward compatibility ?
Also, isn't ":" an allowed character in portlets names ?

**Comment by trongtt** [comment]
From what I see, it doesn't break backward compatibility.

Why I choose ":" as the separator for application and portlet names ? It's because ":" is not allowed in application name (war file name). So first part will be application name, and second part (even including another ":") will be portlet name.

**Comment by thomasdelhomenie** [comment]
ok, thanks @trongtt

**Comment by trongtt** [comment]
Merged into develop

**Pull request closed without merging**